client/components/social-buttons/google.js

```diff
@@ -140,6 +141,15 @@ class GoogleLoginButton extends Component {
 
 	handleClick( event ) {
 		event.preventDefault();
+		event.stopPropagation();
+
+		if ( this.state.error && this.state.eventTimeStamp !== event.timeStamp ) {
+			this.setState( {
+				showError: ! this.state.showError,
```

Comment:
I know this used to be like that before, but it's worth fixing while we're here: directly relying on state values for calculating the next state can lead to a potentially inconsistent value and subtle bugs. This is because React optimizes for performance and may batch several `setState()` into a single update. 

Fixing this is as easy as replacing the `this.setState()` call to use the callback form:

```
this.setState( ( state ) => ( {
	showError: ! state.showError,
	errorRef: event.currentTarget,
	eventTimeStamp: event.timeStamp,
} ) );
```

You can read more about async state updates here: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous